thavalon-webapp/src/components/Register.tsx

```diff
@@ -87,43 +102,20 @@ export function Register(props: RegisterProps) {
             className="Modal"
             overlayClassName="Overlay"
         >
-            <h2 className="modalHeader">Register</h2>
-            <form onSubmit={handleSubmit(onSubmit)}>
-                <input
-                    type="text"
-                    placeholder="Name"
-                    name="name"
-                    ref={register({required: true})}
-                    />
-                {errors.name && <span className="errorMsg">Name required.</span>}
-                <br />
-                <input
-                    type="email"
-                    placeholder="Email"
-                    name="email"
-                    ref={register({required: true, maxLength: 80, pattern: {
-                        value: /^\S+@\S+\.\S+$/i,
-                        message: "Invalid email address."
-                    }})} />
-                {errors.email && <span className="errorMsg">{errors.email.message}</span>}
-                <br />
-                <input
-                    type="password"
-                    placeholder="Password"
-                    name="password"
-                    ref={register({required: true, minLength: 8})} />
-                {errors.password && <span className="errorMsg">{errors.password.message}.</span>}
-                <br />
-                <input
-                    type="password"
-                    placeholder="Confirm Password"
-                    name="confirmPassword"
-                    ref={register({required: true, minLength: 8})} />
-                {errors.confirmPassword && <span className="errorMsg">{errors.confirmPassword.message}</span>}
-                <br />
-                <input type="submit" disabled={disable} value="Register" />
-                <span className="errorMsg">{formErrorMsg}</span>
-            </form>
+            <div className="modalContainer">

```

Comment:
tioli, but this code is very similar to Login.tsx. If we think this is going to be used elsewhere, it might be worth an abstraction to a new component that takes a list of `InpuElements`, a button label and config, and an optional `Link` to redirect. It would be more work though, so if this and Login.tsx is the only place we'll use this code it's probably fine.

Reply:
I think that's an unnecessary layer of abstraction. There's minimal benefit, and if we ever want to do something specific for just login or register we'd have to revert back